test/jacobian.jl

```diff
@@ -26,4 +26,52 @@ end
     else
         @test J(current_state(ds), current_parameters(ds), 0.0) == result
     end
-end
\ No newline at end of file
+end
+
+@testset "MTK Jacobian" begin
+    using ModelingToolkit
+    using ModelingToolkit: Num, RuntimeGeneratedFunctions.RuntimeGeneratedFunction
+    using DynamicalSystemsBase: SciMLBase
+    @independent_variables t
+    @variables u(t)[1:2]
+    D = Differential(t)
+    p = 3.0
+
+    eqs = [D(u[1]) ~ p * u[1],
+        D(u[2]) ~ -p * u[2]]
+
+    @named sys = ODESystem(eqs, t)
+    sys = structural_simplify(sys)
+
+    jac = calculate_jacobian(sys)
+    @test jac isa Matrix{Num}
+
+    prob = ODEProblem(sys, [1.0, 1.0], (0.0, 1.0); jac=true)
+    ode = CoupledODEs(prob)
+    @test ode.integ.f.jac.jac_oop isa RuntimeGeneratedFunction
+    @test ode.integ.f.jac([1.0, 1.0], [3.0], 0.0) isa Matrix{Float64}
+
+    jac = jacobian(ode)
+    @test jac.jac_oop isa RuntimeGeneratedFunction
+    @test jac([1.0, 1.0], [3.0], 0.0) isa Matrix{Float64}
+
+    @testset "CoupledSDEs" begin
+        using StochasticDiffEq
+        @brownian β
+        eqs = [D(u[1]) ~ p * u[1]+ β,
+                D(u[2]) ~ -p * u[2] + β]
+        @mtkbuild sys = System(eqs, t)
+
+        jac = calculate_jacobian(sys)
+        @test jac isa Matrix{Num}
+
+        prob = SDEProblem(sys, [1.0, 1.0], (0.0, 1.0), jac=true)
+        sde = CoupledSDEs(prob)
+        @test sde.integ.f.jac.jac_oop isa RuntimeGeneratedFunction
+        @test sde.integ.f.jac([1.0, 1.0], [3.0], 0.0) isa Matrix{Float64}
+
+        jac = jacobian(ode)
+        @test jac.jac_oop isa RuntimeGeneratedFunction
```

Comment:
Generally speaking testing the type of things is not of much value. We test functionality and interfaces. Instead of testing whether something "is what it should be", test instead that it "does what it should do".